From: Sunita Nadampalli <sunitan@ti.com>
Date: Thu, 3 Feb 2011 16:41:58 +0000 (-0600)
Subject: V4L2 changes for controlling z-order
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=c7c8b70f523b202adfdb9b2c98d41228f77b5227;hp=df77fb42a1378d84c61bbb2b8cdbaecd0fd2a4bd

V4L2 changes for controlling z-order

Description# v4l2 window structure is extended to take z-order for the
window.
The default assignment for the video pipes is modified to assign the
highest value (zorder 3) for graphics plane and lowest for video pipes.
Overlay format set and get are modified to configure the z-order.

Change-Id: I0ba3ee95cee9b307cbc4dd2502ad86f4f30c11db
Signed-off-by: Sunita Nadampalli <sunitan@ti.com>
---

Index: Kernel/drivers/media/video/omap/omap_vout.c
===================================================================
--- Kernel.orig/drivers/media/video/omap/omap_vout.c	2011-02-07 18:55:20.000000000 +0530
+++ Kernel/drivers/media/video/omap/omap_vout.c	2011-02-08 17:07:57.547781888 +0530
@@ -2300,6 +2300,8 @@
 				info.out_height = vout->win.w.height;
 				info.out_width = vout->win.w.width;
 			}
+			if (cpu_is_omap44xx())
+			        info.zorder = vout->win.zorder;
 
 			if (ovl->set_overlay_info(ovl, &info))
 				return -EINVAL;
@@ -2345,6 +2347,8 @@
 	win->w = vout->win.w;
 	win->field = vout->win.field;
 	win->global_alpha = vout->win.global_alpha;
+        if (cpu_is_omap44xx())
+            win->zorder = vout->win.zorder;
 
 	if (ovl->manager && ovl->manager->get_manager_info) {
 		ovl->manager->get_manager_info(ovl->manager, &info);
Index: Kernel/drivers/media/video/omap/omap_voutlib.c
===================================================================
--- Kernel.orig/drivers/media/video/omap/omap_voutlib.c	2011-02-07 18:55:20.000000000 +0530
+++ Kernel/drivers/media/video/omap/omap_voutlib.c	2011-02-08 17:03:35.909018437 +0530
@@ -119,6 +119,8 @@
 	win->w = new_win->w;
 	win->field = new_win->field;
 	win->chromakey = new_win->chromakey;
+        if (cpu_is_omap44xx())
+            win->zorder = new_win->zorder;
 
 	/* Adjust the cropping window to allow for resizing limitation */
 	if (cpu_is_omap24xx()) {
Index: Kernel/drivers/video/omap2/dss/overlay.c
===================================================================
--- Kernel.orig/drivers/video/omap2/dss/overlay.c	2011-02-07 18:56:28.000000000 +0530
+++ Kernel/drivers/video/omap2/dss/overlay.c	2011-02-08 17:03:35.919022851 +0530
@@ -747,7 +747,7 @@
 				OMAP_DSS_COLOR_GFX_OMAP2;
 			ovl->caps = OMAP_DSS_OVL_CAP_DISPC;
 			ovl->info.global_alpha = 255;
-			ovl->info.zorder = OMAP_DSS_OVL_ZORDER_0;
+			ovl->info.zorder = OMAP_DSS_OVL_ZORDER_3;
 			break;
 		case 1:
 			ovl->name = "vid1";
@@ -762,7 +762,7 @@
 				OMAP_DSS_COLOR_CONV_BT601_5_LR;
 			ovl->info.yuv2rgb_conv.dirty = true;
 			ovl->info.global_alpha = 255;
-			ovl->info.zorder = OMAP_DSS_OVL_ZORDER_3;
+			ovl->info.zorder = OMAP_DSS_OVL_ZORDER_2;
 			break;
 		case 2:
 			ovl->name = "vid2";
@@ -777,7 +777,7 @@
 				OMAP_DSS_COLOR_CONV_BT601_5_LR;
 			ovl->info.yuv2rgb_conv.dirty = true;
 			ovl->info.global_alpha = 255;
-			ovl->info.zorder = OMAP_DSS_OVL_ZORDER_2;
+			ovl->info.zorder = OMAP_DSS_OVL_ZORDER_1;
 			break;
 		case 3:
 			ovl->name = "vid3";
@@ -789,7 +789,7 @@
 				OMAP_DSS_COLOR_CONV_BT601_5_LR;
 			ovl->info.yuv2rgb_conv.dirty = true;
 			ovl->info.global_alpha = 255;
-			ovl->info.zorder = OMAP_DSS_OVL_ZORDER_1;
+			ovl->info.zorder = OMAP_DSS_OVL_ZORDER_0;
 		}
 
 		ovl->info.min_x_decim = ovl->info.min_y_decim = 1;
Index: Kernel/include/linux/videodev2.h
===================================================================
--- Kernel.orig/include/linux/videodev2.h	2011-02-07 18:57:07.000000000 +0530
+++ Kernel/include/linux/videodev2.h	2011-02-08 17:03:35.919022851 +0530
@@ -589,6 +589,7 @@
 	__u32			clipcount;
 	void			__user *bitmap;
 	__u8                    global_alpha;
+        __u8                    zorder;
 };
 
 /*
